.github/workflows/test-aks.yaml

```diff
@@ -0,0 +1,93 @@
+  pull_request:
+    paths:
+      - tasks/iac.yaml
+      - .github/bundles/aks/*
+      - .github/test-infra/azure/aks/*
+
+permissions:
+  id-token: write
+  contents: read
+
+jobs:
+  test-aks-install:
+    strategy:
+      matrix:
+        flavor: [upstream, registry1, unicorn]
+    runs-on: ubuntu-latest
+    env:
+      SHA: ${{ github.sha }}
+      UDS_REGION: centralus
+      UDS_RESOURCE_GROUP_NAME: ${{ secrets.AZURE_RESOURCE_GROUP }}
+      UDS_STORAGE_ACCOUNT_NAME: ${{ secrets.AZURE_STORAGE_ACCOUNT_NAME }}
+      UDS_CONTAINER_NAME: ${{ secrets.AZURE_STORAGE_CONTAINER_NAME }}
+
+    steps:
+      - name: Set ENV
+        run: |
+          echo "UDS_CLUSTER_NAME=uds-ci-${{ matrix.flavor }}-${SHA:0:7}" >> $GITHUB_ENV
+          echo "UDS_STATE_KEY="uds-core/${SHA:0:7}-nightly-aks-core-${{ matrix.flavor }}.tfstate >> $GITHUB_ENV
+          echo "TF_VAR_location=${UDS_REGION}" >> $GITHUB_ENV
+          echo "TF_VAR_cluster_name=uds-ci-${{ matrix.flavor }}-${SHA:0:7}" >> $GITHUB_ENV
+          echo "TF_VAR_resource_group_name=uds-ci-${{ matrix.flavor }}" >> $GITHUB_ENV
+          echo "ARM_SUBSCRIPTION_ID=${{ secrets.AZURE_SUBSCRIPTION_ID }}" >> $GITHUB_ENV
+          echo "ARM_CLIENT_ID=${{ secrets.AZURE_CLIENT_ID }}" >> $GITHUB_ENV
+          echo "ARM_TENANT_ID=${{ secrets.AZURE_TENANT_ID }}" >> $GITHUB_ENV
+          echo "ARM_USE_OIDC=true" >> $GITHUB_ENV
+          echo "ARM_STORAGE_USE_AZUREAD=true" >> $GITHUB_ENV
+
+      - name: Checkout repository
+        uses: actions/checkout@eef61447b9ff4aafe5dcd4e0bbf5d482be7e7871 # v4
+
+      - name: Azure login
+        uses: azure/login@v2
+        with:
+          client-id: ${{ secrets.AZURE_CLIENT_ID }}
+          tenant-id: ${{ secrets.AZURE_TENANT_ID }}
+          subscription-id: ${{ secrets.AZURE_SUBSCRIPTION_ID }}
+
+      - name: Environment setup
+        uses: ./.github/actions/setup
+        with:
+          registry1Username: ${{ secrets.IRON_BANK_ROBOT_USERNAME }}
+          registry1Password: ${{ secrets.IRON_BANK_ROBOT_PASSWORD }}
+          ghToken: ${{ secrets.GITHUB_TOKEN }}
+          chainguardIdentity: ${{ secrets.CHAINGUARD_IDENTITY }}
+
+      - name: Setup Tofu
+        uses: opentofu/setup-opentofu@12f4debbf681675350b6cd1f0ff8ecfbda62027b # v1.0.4
+        with:
+          tofu_version: 1.8.2
+          tofu_wrapper: false
+
+      - name: Create UDS Core Package
+        run: ZARF_ARCHITECTURE=amd64 uds run -f tasks/create.yaml standard-package --no-progress --set FLAVOR=${{ matrix.flavor }}
+
+      - name: Create Core Bundle
+        run: uds create .github/bundles/aks --confirm
+
+      - name: Create IAC
+        run: uds run -f tasks/iac.yaml apply-tofu --no-progress --set K8S_DISTRO=aks --set CLOUD=azure
+
+      - name: Deploy Core Bundle
+        env:
+          UDS_CONFIG: .github/bundles/aks/uds-config.yaml
+        run: uds deploy .github/bundles/aks/uds-bundle-uds-core-aks-nightly-*.tar.zst --confirm
+        timeout-minutes: 30
```

Comment:
It would be nice after the core deployment (in each cloud/distro) to run our [debug output action](https://github.com/defenseunicorns/uds-core/blob/main/.github/actions/debug-output/action.yaml). That would give a bit more confidence without seeing the cluster that everything is running as expected, and provide a quick debug sanity check if this fails in the future. You should be able to just drop in this snippet, before the remove:

```yaml
      - name: Debug Output
        if: ${{ always() }}
        uses: ./.github/actions/debug-output
```

Reply:
(noting that this isn't directly connected to the AKS work, but given my lack of visibility into AKS it would be helpful in reviewing this to get a quick scan of everything up and running)

Reply:
added in https://github.com/defenseunicorns/uds-core/pull/908/commits/a5af9d3936803dec6d68cf9c74cf7c14ec3c29ae